test/unit/control-connection-tests.js

```diff
@@ -58,6 +58,21 @@ describe('ControlConnection', function () {
         done();
       });
     });
+    it('should resolve IPv4 and IPv6 addresses with non default port', function (done) {
+      if (!useLocalhost) {
+        return done();
+      }
+      var cc = new ControlConnection(clientOptions.extend({ contactPoints: ['localhost:9999'] }));
+      cc.getConnection = helper.callbackNoop;
+      cc.refreshOnConnection = helper.callbackNoop;
+      cc.init(function (err) {
+        assert.ifError(err);
+        var hosts = cc.hosts.values();
+        assert.strictEqual(hosts.length, 2);
```

Comment:
The test is still failing because the `hosts.length` can be 1 or 2 depending on the system and because `hosts.indexOf('127.0.0.1:9999') >= 0` can not be `true` as `hosts` is an `Array<Host>`. Use [`Host#address`](http://docs.datastax.com/en/developer/nodejs-driver/3.2/api/class.Host/#address) instead.